nupic/torch/compatibility.py

```diff
@@ -0,0 +1,47 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from collections import OrderedDict
+
+import torch
+
+
+def upgrade_to_masked_sparseweights(state_dict):
```

Comment:
This function is not called here. Maybe you could use `_register_load_state_dict_pre_hook` in `SparseWeightsBase ` to replace the keys before `load_state_dict` is called

Reply:
I played around with this. The problem is that it will require mutating the state dict that is passed in, which could cause issues. I also considered overriding `_load_from_state_dict`, but that would fail when `strict=False` (unless I mutate the state_dict).

`upgrade_to_masked_sparseweights` is called in nupic.research, by the way.